packages/angular_devkit/core/src/json/schema/registry.ts

```diff
@@ -103,30 +112,40 @@ export class CoreSchemaRegistry implements SchemaRegistry {
 
     this._ajv = ajv({
       formats: formatsObj,
-      loadSchema: (uri: string) => this._fetch(uri),
+      loadSchema: (uri: string) => this._fetch(uri).toPromise(),
       schemaId: 'auto',
       passContext: true,
     });
 
     this._ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-04.json'));
     this._ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-06.json'));
-
-    this.addPostTransform(addUndefinedDefaults);
   }
 
-  private _fetch(uri: string): Promise<JsonObject> {
+  private _fetch(uri: string): Observable<JsonObject> {
```

Comment:
using a promise seems to make more sense here since the observable is being created to then be turned immediately into a promise.

Actually and more generally, as an experiment, I converted the compile/validate method to use async/await instead of observables.  It's shorter, easier to read, and actually debuggable.  With the public API remaining observable-based and consuming code needing to assume that the validating function is asynchronous anyway, I'm not sure there is a compelling reason to use observables internally here.

Reply:
It is, but Observables are cancellable and have other advantages. I say that even for our private stuff we should use Promises carefully.

I changed back `_fetch()`; I was using it in flatten at some point but now I just use ajv's logic.